src/vcpkg/base/downloads.cpp

```diff
@@ -253,12 +378,17 @@ namespace vcpkg
             if (auto p_https_proxy = maybe_https_proxy_env.get())
             {
                 std::wstring env_proxy_settings = Strings::to_utf16(*p_https_proxy);
+                ProxyUrlParts parts = parse_proxy_url(env_proxy_settings);
```

Comment:
I think it would be a good idea to make the parsing stuff etc. use chars rather than wchar_ts making that parsing code consistent with all other parsing code in the product.